### ISSUE TITLE
sbt-ethereum workshop 2019-10-13

### ISSUE THREAD

**Issue by swaldman** [swaldman/eth-quip-client#1]
This issue is intended to host conversation (and help us share Ethereum addresses) during the Ethereum Silicon Valley workshop on sbt-ethereum, 2019-10-13.

Note that the address of the `Quip` contract that will be the center of our exercise is `0x2a03eb37c0077dba0814a004bc53f2567b7587a0`. (You can find it on Etherscan [here](https://etherscan.io/address/0x2a03eb37c0077dba0814a004bc53f2567b7587a0).)

Here is the public interface of that contract:
```solidity
contract Quip {

  // events
  
  event QuipAdded( address indexed quipper, string quip );
  event VoterRegistered( address indexed voter );
  event VoteUpdated( address indexed voter, address indexed quipper, uint256 indexed quipIndex, string quip );
  event QuipPaid( address indexed voter, address indexed quipper, uint256 indexed quipIndex, string quip, address payoutToken, uint256 payout );

  // public accessors

  function quipCount()                  public view returns( uint256 count );
  function getQuip(uint256 i)           public view returns( string memory quip, address quipper )
  function hasVoted( address voter )    public view returns ( bool voted );
  function currentVote( address voter ) public view returns( uint256 index );
  function uniformRandomQuip()          public view returns( string memory quip, address quipper );
  function voteWeightedRandomQuip()     public view returns( string memory quip, address quipper );

  // public mutators

  function addQuip( string memory quip )           public returns( uint256 index );
  function vote( uint256 index )                   public;
  function payout( address token, uint256 amount ) public payable;
  
}
```

**Comment by garyjob** [comment]
Gary here: 0x240ac70cb56912f3ab7880e8caf14c1be3581d4b
Give me some mula!! Boomz!

**Comment by adrianpeng1992** [comment]
0xf1b5f894277f6134722f27fd013c3cf6fa059527

**Comment by adnjoo** [comment]
hello 0x2b3aae22ed29b5b967d9f0e389c0a03f8d98247d

**Comment by cavepainting** [comment]
AJ Eth Address: 0xD860Aa176040aDE73224f3D7a02948096eE4a98b

**Comment by dchem** [comment]
0x0dd58914454901cf811e30c809c99e6068f5b435

**Comment by swaldman** [comment]
DAI address: 0x89d24a6b4ccb1b6faa2625fe562bdd9a23260359

**Comment by chrisvwx** [comment]
Thanks Steve!

**Comment by swaldman** [comment]
Thank you Chris for all the organizing!

**Comment by swaldman** [comment]
Thank you all for coming. I really enjoyed this event, despite being embarrassed by some glitches. Those glitches are now fixed, in sbt-ethereum v0.3.3. If you want to upgrade, either...

1) Delete the `eth-command-line` directory you made and re-clone it with `git clone https://github.com/swaldman/eth-command-line.git --branch 0.3.3` (see https://www.sbt-ethereum.io/tutorials/getting-started.html for a better formatted version of that). Deleting this directory won't delete your wallets or funds or anything like that.

or, alternatively,

2) In your `eth-command-line` directory, there is a subdirectory called `project`. In that directory, there is a file called `plugins.sbt`. Edit that to change the version number from 0.3.2 to 0.3.3

I am very glad to help people with sbt-ethereum. You can get in touch by e-mail (please put sbt-ethereum in the subject for a faster response!) at swaldman@mchange.com.

My preferred support venue is https://ethereum.stackexchange.com/ Mark your question with the tag "sbt-ethereum" and I will see it.

If you are in the area and want more help in person, individually or in groups, let me know and we'll set something up.

Thank you all again for coming and helping me make sbt-ethereum a bit better. I do hope that you find it a fun and useful tool.
